### PR TITLE
feat(search): reverse image search

### PR DESCRIPTION
**Pull request by gkartalis**
The type of this PR is: **TYPE**

This PR resolves []

### PR Checklist (tick all before merging)

- [ ] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [ ] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [ ] I have documented any follow-up work that this PR will require, or it does not require any.
- [ ] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [ ] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add experimental reverse image search to identify artworks (behind feature flag) - gkartalis

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4445ff9dbdd0d4a02440234b1278b0abcfe4d8fe
</p>

**Review by dblandin — APPROVED** [review]
This LGTM. I left a few comments but nothing blocking.

File: src/palette/elements/Input/Input.tsx [hunk 1]
```diff
@@ -297,6 +310,13 @@ export const Input = React.forwardRef<TextInput, InputProps>(
                 }}
               />
             </Flex>
+            {!!showCamera && (
```
[the remark this answers is not in the document]
Comment (gkartalis) [reply src/palette/elements/Input/Input.tsx:313]: 🤔 I thought of that as well but in the end decided to just introduce this `showCamera: boolean` flag to hide all the camera related functionality behind this prop. 

With this approach this is hidden everywhere where we don't need it.

Maybe in the future we could move it to a new `<SearchInput />` component but for now I don't see a reason. What do you think?

[the remark this answers is not in the document]
Comment (dblandin) [reply src/palette/elements/Input/Input.tsx:313]: Yep, I don't think we need to address it now but would be a nice follow-up to look into.

**Pull request merged**